### ISSUE TITLE
S1121 possible false positive when using with expressions on records

### ISSUE THREAD

**Issue by rubiktubik** [SonarSource/sonar-dotnet#4417]
### Description
I'am getting a S1121 with SonarLint in Visual-Studio when using a with-expression with records

### Repro steps

```cs

myState = myState with
        {
		Value = 12m,
        }
```

### Expected behavior

No Warning

### Actual behavior

Getting a S1121 Warning with "Extract the assignment of 'Value' from this expression."

### Known workarounds

Disable the rule

### Related information

* C#/VB.NET Plugins version - SonarLint 4.3
* Visual Studio version - Visual Studio 2019 16.9.3
* Operating System - Windows 10 - 1909

**Comment by pavel-mikula-sonarsource** [comment]
Hi @rubiktubik ,

Thank you for reporting this issue. I can confirm it as False Positive. And you're right in time for the party, as it was fixed last week on `master` and will be part of the next release 8.23 of our Analyzer.
